### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by lwahlmeier**
large refactory, 

* Commonized API so we can do bb.org or bb server.  
* Fixed/cleaned up how tasks work.  
* Reuses http clients instead of creating them on every request (major speed up).
* Retry on 429
* Get http client using logger
* Moved task/go routine management into taskQueue creator function
* Fix logging of workers
* Moved back to mkdocs 1.0.4.

### CODE REVIEW

File: scripts/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-mkdocs==1.1.2
+mkdocs==1.0.4
```
Comment (thorix) [inline scripts/requirements.txt:1]: Don't versions count up?

**Review by thorix — APPROVED** [review]
:+1:

File: scripts/requirements.txt (hunk 1 above)
Comment (lwahlmeier) [reply scripts/requirements.txt:1]: 1.1.2 causes problems, apparently its not fully compatible with the way we make the mkdocs.yaml file.  Once I get a chance to look at why I will fix it and move it up.

**Review by hyperbolist — APPROVED** [review]
Discussed via voice.

**Pull request merged**